src/backend/distributed/deparser/deparse_database_stmts.c

```diff
@@ -205,3 +224,88 @@ DeparseAlterDatabaseSetStmt(Node *node)
 
 	return str.data;
 }
+
+
+static void
+AppendCreateDatabaseStmt(StringInfo buf, CreatedbStmt *stmt)
+{
+	appendStringInfo(buf,
+					 "CREATE DATABASE %s",
+					 quote_identifier(stmt->dbname));
+
+	DefElem *option = NULL;
+
+	foreach_ptr(option, stmt->options)
+	{
+		/*If option is template, lc_type, locale or lc_collate, propagation will not be supportted */
+		/* since template database is not stored in the catalog */
+		if (strcmp(option->defname, "template") == 0 ||
+			strcmp(option->defname, "strategy") == 0 ||
+			strcmp(option->defname, "lc_ctype") == 0 ||
+			strcmp(option->defname, "locale") == 0 ||
+			strcmp(option->defname, "lc_collate") == 0 ||
+			strcmp(option->defname, "icu_locale") == 0 ||
+			strcmp(option->defname, "locale_provider") == 0)
+		{
+			ereport(ERROR,
+					errmsg("CREATE DATABASE option \"%s\" is not supported",
+						   option->defname));
+		}
```

Comment:
Let's make this block a function `ValidateCreateDatabaseOptions(options)` and call it in PostprocessCreateDatabase to make it explicit what we don't support. And let's call `ValidateCreateDatabaseOptions` here too, to make further re-use of 
`AppendCreateDatabaseStmt` function safe.